### PR TITLE
fix: test: add unit tests for string/path utility edge cases

### LINKED ISSUE #20: test: add unit tests for string/path utility edge cases
## Background

Utility functions are a good low-risk entry point for C++ contributors. Adding edge-case coverage helps contributors learn the test setup without touching the media pipeline, hardware backend, or service lifecycle.

## Scope

Suggested files:

- Existing utility tests under `test/`, such as string/path-related test files
- Related utility implementation files only if a bug is discovered while writing tests

## Acceptance Criteria

- Add focused unit tests for string and/or path utility edge cases.
- Cover cases such as empty input, trailing separators, repeated separators, unusual extensions, or platform-style path separators where relevant.
- Keep the change small and avoid unrelated refactors.
- If behavior is unclear, document the observed behavior in the test name or comments.

## Validation

```bash
bash scripts/build_cpu_test.sh
./build_cpu/cosmo-tests
```

## Suggested Labels

`area/backend`, `area/tests`, `difficulty/easy`

### PR DESCRIPTION
**Pull request by VedantMadane**
## Summary

Add Catch2 unit tests for string/path utility edge cases.

## Changes

- `test/test_string_util.cc` — consecutive/leading delimiters, trailing separators, compound extensions, backslash behavior, JoinStrings/Trim edges
- `test/test_path_util.cc` — IsSafePathComponent length/control-byte edges; IsWithinRoot empty/repeated-separator edges

Fixes #20

### CODE REVIEW

**Review by Fernandez81 — CHANGES_REQUESTED** [review]
Thanks for the contribution. This change does not currently satisfy #20 and cannot be merged as-is:

1. `pathUtil.edge.test.js` imports `packet_stringifier.go` as an ES module. Node rejects the `.go` extension with `ERR_UNKNOWN_FILE_EXTENSION`.
2. That vendored Pion RTCP file contains Go packet-stringification code and does not export `normalizeSlashes`, `isAbsolutePath`, or `joinPath`. As a result, these tests do not exercise any CosmoEdge string/path utility.
3. The first normalization assertion is too permissive: an unchanged value such as `a\\b` still contains `a`, so incorrect behavior would pass. The standalone `test:edge` script is also not part of the C++ test target or any CI command.

Please remove the top-level JavaScript test and the root `package.json` script. Add focused Catch2 cases to the existing `test/test_string_util.cc`, `test/test_path_util.cc`, and/or `test/test_file_util.cc`, using exact expected results from the actual CosmoEdge implementations. Then validate with the repository-supported Docker flow for `scripts/build_cpu_test.sh` and `build_cpu/cosmo-tests`.

**Comment by VedantMadane** [comment]
Hi @Fernandez81, thanks for the guidance!

I have removed the JS test and added focused Catch2 test cases directly to the existing C++ test suite:
- `test/test_string_util.cc`: edge cases for `SplitString` (consecutive/leading delimiters, empty string), `GetFileExtension` (compound extensions, no extension, trailing dot), `GetFileName` (trailing slashes, backslashes), `Trim` (all whitespace, control chars), and `JoinStrings` (single item, empty list).
- `test/test_path_util.cc`: edge cases for `IsSafePathComponent` (length limits, control bytes) and `IsWithinRoot` (empty paths, repeated separators).

Cleanly rebased on latest `main`.

File: test/test_path_util.cc [hunk 1]
```diff
@@ -254,7 +255,7 @@ TEST_CASE("PathUtil: strict root resolution fails closed", "[path-util][security
 }
 
 TEST_CASE("PathUtil: path component validation", "[path-util][security]") {
-    REQUIRE(IsSafePathComponent("task-01_中文"));
+    REQUIRE(IsSafePathComponent("task-01_ä¸­æ–‡"));
```
Comment (Fernandez81) [inline test/test_path_util.cc:259]: Looks like the Chinese characters got garbled here. Could you change this back to task-01_中文?

File: _pr_meta.json
Comment (Fernandez81) [inline _pr_meta.json:1]: Remove the PR metadata file.

**Review by Fernandez81 — APPROVED** [review]
